src/realm/query.cpp

```diff
@@ -1441,6 +1436,78 @@ size_t Query::count(size_t start, size_t end, size_t limit) const
     return cnt;
 }
 
+size_t Query::count(size_t start, size_t end, size_t limit) const
+{
+#if REALM_METRICS
+    std::unique_ptr<MetricTimer> metric_timer = QueryInfo::track(this, QueryInfo::type_Count);
+#endif
+    return do_count(start, end, limit);
+
+}
+
+TableView Query::find_all(const DescriptorOrdering& descriptor)
+{
+#if REALM_METRICS
+    std::unique_ptr<MetricTimer> metric_timer = QueryInfo::track(this, QueryInfo::type_FindAll);
+#endif
+    const size_t default_start = 0;
+    const size_t default_end = size_t(-1);
+    const size_t default_limit = size_t(-1);
+
+    if (descriptor.is_empty()) {
+        TableView ret(*m_table, *this, default_start, default_end, default_limit);
+        find_all(ret);
+        return ret;
+    }
+
+    bool only_limit = true;
+    size_t min_limit = size_t(-1);
+    for (size_t i = 0; i < descriptor.size(); ++i) {
+        if (!descriptor.descriptor_is_limit(i)) {
+            only_limit = false;
+            break;
+        } else {
+            const LimitDescriptor* limit = dynamic_cast<const LimitDescriptor*>(descriptor[i]);
+            REALM_ASSERT(limit);
+            min_limit = std::min(min_limit, limit->get_limit());
+        }
+    }
+    if (only_limit) {
+        TableView ret(*m_table, *this, default_start, default_end, default_limit);
```

Comment:
TableView should be initialized with `min_limit`. Otherwise `sync_if_needed` will not give desired result.

Reply:
Good catch! I fixed it and added a test to cover that case.